File: src/main/java/es/santander/ascender/retoGrupoCIC/model/Pelicula.java
package es.santander.ascender.retoGrupoCIC.model;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.MapsId;
import jakarta.persistence.OneToOne;

@Entity
public class Pelicula {
    @Id
    private Long itemId;

    private String director;
    private Integer duracion; // Duración en minutos
    private String genero;
    private LocalDate fechaEstreno;

    @OneToOne
    @MapsId
    private Item item;

}

File: front/src/app/models/pelicula.ts
// src/pelicula.ts
export interface Pelicula {
  id: number;
  director: string;
  duracion: number;
  genero: string;
  fechaEstreno: string;
  item: {
    id: number;
    nombre: string;
    ubicacion: string;
    tipo: {
      id: number;
      nombre: string;
    };
    formato: {
      id: number;
      nombre: string;
    };
    fecha: string;
    estado: string;
  };
}
